lib/IssueBodyChecker.js

```diff
@@ -0,0 +1,81 @@
+function fromBase64 (content) {
+  return Buffer.from(content, 'base64').toString()
+}
+
+async function getIssueTemplateBody (context, path) {
+  try {
+    const singleTemplate = await context.github.repos.getContent({
+      owner: context.payload.repository.owner.login,
+      repo: context.payload.repository.name,
+      path
+    })
+
+    return fromBase64(singleTemplate.data.content)
+  } catch (err) {
+    return null
+  }
+}
+
+async function getIssueTemplatePaths (context) {
+  try {
+    const templatesDirectory = await context.github.repos.getContent({
+      owner: context.payload.repository.owner.login,
+      repo: context.payload.repository.name,
+      path: '.github/ISSUE_TEMPLATE/'
+    })
+
+    return templatesDirectory.data.map(f => f.path)
+  } catch (err) {
+    return null
+  }
+}
+
+async function getIssueTemplates (context) {
+  const defaultTemplate = await getIssueTemplateBody(context, '.github/ISSUE_TEMPLATE.md')
+  if (defaultTemplate !== null) {
+    return [defaultTemplate]
+  }
+
+  const paths = await getIssueTemplatePaths(context)
+  if (paths !== null) {
+    const templates = []
+    for (const path of paths) {
+      const template = await getIssueTemplateBody(context, path)
+      if (template != null) {
+        templates.push(template)
+      }
+    }
+
+    return templates
+  }
+
+  return []
+}
+
+async function isBodyEqualToTemplate (body, context) {
+  const issueTemplates = await getIssueTemplates(context)
+
+  for (const template of issueTemplates) {
+    if (body.includes(template)) {
```

Comment:
This is my main concern with this PR. What if the template was:
```
Title:
Description:
Issue:
```

and I opened an issue with:
```
Title: blah blah
Description: blah blah blah
Issue: more blah
```

This check will fail. If I'm following correctly, `body` is the string of the issue body. Which will be something like:

`Title: blah blah\nDescription: blah blah blah\nIssue: more blah` 

which will not ["include"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes):

`Title:\nDescription:\nIssue:`

`includes()` as a string method is very literal and doesn't handle the sort of pattern matching or regex that we're looking for here. I think someone opening an issue with a the _exact_ contents of an issue template (ie not filled in) is equally bad to it being blank.

Reply:
The checker at the bottom of the file returns `false` (i.e. invalid) if this function returns `true`. This checks that the OP has actually filled out the template and not just left it blank.

Reply:
The check I'm trying to do here is whether the issue body matches a template defined in the repository. I want to catch and handle those cases that @j-f1 calls out - people who just submit the template as-is, without making any changes to the issue body. I'm not interested in doing more advanced checks (like trying to correlate fields and values) currently.